controller/controllerutil.go

```diff
@@ -0,0 +1,78 @@
+package controller
+
+/*
+Copyright 2017 - 2020 Crunchy Data Solutions, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	crv1 "github.com/crunchydata/postgres-operator/apis/cr/v1"
+	"github.com/crunchydata/postgres-operator/config"
+	"github.com/crunchydata/postgres-operator/kubeapi"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/rest"
+)
+
+// SetClusterInitializedStatus sets the status of a pgcluster CR to indicate that it has been
+// initialized.  This is specifically done by patching the status of the pgcluster CR with the
+// proper initialization status.
+func SetClusterInitializedStatus(restclient *rest.RESTClient, clusterName,
+	namespace string) error {
+
+	cluster := crv1.Pgcluster{}
+	if _, err := kubeapi.Getpgcluster(restclient, &cluster, clusterName,
+		namespace); err != nil {
+		log.Error(err)
+		return err
+	}
+	message := "Cluster has been initialized"
+	if err := kubeapi.PatchpgclusterStatus(restclient, crv1.PgclusterStateInitialized, message,
+		&cluster, namespace); err != nil {
+		log.Error(err)
+		return err
+	}
+
+	return nil
+}
+
+// InitializeReplicaCreation initializes the creation of replicas for a cluster.  For a regular
+// (i.e. non-standby) cluster this is called following the creation of the initial cluster backup,
+// which is needed to bootstrap replicas.  However, for a standby cluster this is called as
+// soon as the primary PG pod reports ready and the cluster is marked as initialized.
+func InitializeReplicaCreation(restclient *rest.RESTClient, clusterName,
```

Comment:
I'd recommend putting this function up top so it's in alphabetical order. easier for browsing when knowing what to expect. I know in older files, it's been harder to enforce this.

Reply:
Moved `InitializeReplicaCreation` to the top.